drivers/console/bm_serial.c

```diff
@@ -275,22 +288,28 @@ static void bm_serial_isr(const struct device *dev, void *user_data)
 	}
 }
 
-static void bm_serial_setup(const struct device *uart)
+static void bm_serial_setup(void)
 {
 	uint8_t c;
+	uint8_t i;
 
-	uart_irq_rx_disable(uart);
-	uart_irq_tx_disable(uart);
-
-	/* Drain the fifo */
-	while (uart_fifo_read(uart, &c, 1)) 
+	for ( i = 0; i < MAX_SERIAL_DEV_COUNT; i++ )
 	{
-		continue;
-	}
+		if (upipe_dev[i] != NULL)
+		{
+			uart_irq_rx_disable(upipe_dev[i]);
+			uart_irq_tx_disable(upipe_dev[i]);
```

Comment:
Why do we need to disable the RX interrupt? At the least, we shouldn't need to disable the TX interrupt, but let me know if I'm missing something.